### ISSUE TITLE
[v2] Page hot reload error

### ISSUE THREAD

**Issue by porfirioribeiro** [gatsbyjs/gatsby#6382]
## Description

After this merge https://github.com/gatsbyjs/gatsby/pull/6226
The way as pages are stored in redux changed, so this line
https://github.com/gatsbyjs/gatsby/blob/d1423b1f9c1f9b604c28ad8b31bd094daf13a4b8/packages/gatsby/src/bootstrap/page-hot-reloader.js#L58

Gives error

### Steps to reproduce
Edit any file so it needs to hot reload

### Expected result

Reload the page content

### Actual result

```
error UNHANDLED REJECTION


  TypeError: store.getState(...).pages.filter is not a function

  - page-hot-reloader.js:65
    [website]/[gatsby]/dist/bootstrap/page-hot-reloader.js:65:28
```

**Comment by porfirioribeiro** [comment]
I think here it might give errors

https://github.com/gatsbyjs/gatsby/blob/d1423b1f9c1f9b604c28ad8b31bd094daf13a4b8/packages/gatsby/src/utils/websocket-manager.js#L40

https://github.com/gatsbyjs/gatsby/blob/d1423b1f9c1f9b604c28ad8b31bd094daf13a4b8/packages/gatsby-plugin-page-creator/src/gatsby-node.js#L70

**Comment by m-allanson** [comment]
@pieh's fix was published in `gatsby@2.0.0-beta.23`
